quinn-h3/src/connection.rs

```diff
@@ -267,15 +267,9 @@ impl ConnectionInner {
             .pending_uni
             .iter_mut()
             .enumerate()
-            .filter_map(|(i, x)| {
-                let mut pending = x.take().unwrap();
-                match Pin::new(&mut pending).poll(cx) {
-                    Poll::Ready(y) => Some((i, y)),
-                    Poll::Pending => {
-                        std::mem::replace(x, Some(pending));
-                        None
-                    }
-                }
+            .filter_map(|(i, x)| match Pin::new(x.as_mut().unwrap()).poll(cx) {
+                Poll::Ready(y) => Some((i, y)),
```

Comment:
The previous code left `x` as `None` on `Poll::Ready`. Should this be preserved? If not, should `self.pending_uni` still contain `Option`? Not familiar with this code in general, but that seems suspect.

Reply:
Oops, you're right. Fixed that up.